plans/convert.pp

```diff
@@ -306,9 +306,13 @@
     # Restart cluster compiler services that are likely not restarted by our
     # final Puppet run to increase chance everything is functional upon plan
     # completion
-    run_command('systemctl restart pe-puppetserver.service pe-puppetdb.service',
+    run_command('systemctl restart pe-puppetserver.service',
     $all_targets - $primary_target - $primary_postgresql_target - $replica_postgresql_target)
 
+    # PuppetDB is only found on modern compilers, not legacy ones
+    run_command('systemctl restart pe-puppetdb.service',
+    $all_targets - $primary_target - $primary_postgresql_target - $replica_postgresql_target - $legacy_compiler_targets)
```

Comment:
We have compiler_targets, think it would look cleaner to just use them plus legacy_compiler_targets in these two places. Just wondering if we need this to run on replica_target as well

Reply:
Yeah. Based on the comment, it's not intended to restart the replica, and I assume if the primary doesn't need restarted then neither does the replica. I'll push it up and give it a spin in CI.